filament/src/HwRenderPrimitiveFactory.cpp

```diff
@@ -89,6 +89,17 @@ RenderPrimitiveHandle HwRenderPrimitiveFactory::create(DriverApi& driver,
         // create the backend object
         auto handle = driver.createRenderPrimitive(vbh, ibh,
                 type, offset, minIndex, maxIndex, count);
+
+        // First, check if the handle is already in our set. This shouldn't normally happen, except
+        // when the driver always returns the same handle for render primitives (such as the
+        // NoopDriver).
+        auto pos = mMap.find(handle.getId());
+        if (UTILS_UNLIKELY(pos != mMap.end())) {
+            auto ipos = pos->second;
+            ipos->refs++;
+            return ipos->handle;
+        }
+
```

Comment:
would it work instead to have the noop driver return a random number or a counter? I don't like that this adds overhead in the real case.